core/dbt/config/runtime.py

```diff
@@ -189,28 +205,31 @@ def _get_rendered_profile(
         return Profile.render_from_args(args, profile_renderer, profile_name)
 
     @classmethod
-    def collect_parts(cls: Type["RuntimeConfig"], args: Any) -> Tuple[Project, Profile]:
-        # profile_name from the project
-        project_root = args.project_dir if args.project_dir else os.getcwd()
-        version_check = bool(flags.VERSION_CHECK)
-        partial = Project.partial_load(project_root, verify_version=version_check)
-
+    def get_profile(
+        cls: Type["RuntimeConfig"], args: Any, cli_vars: Dict[str, Any], raw_profile_name: str
+    ):
         # build the profile using the base renderer and the one fact we know
         # Note: only the named profile section is rendered. The rest of the
         # profile is ignored.
-        cli_vars: Dict[str, Any] = parse_cli_vars(getattr(args, "vars", "{}"))
         profile_renderer = ProfileRenderer(cli_vars)
-        profile_name = partial.render_profile_name(profile_renderer)
+        profile_name = profile_renderer.render_value(raw_profile_name)
         profile = cls._get_rendered_profile(args, profile_renderer, profile_name)
         # Save env_vars encountered in rendering for partial parsing
         profile.profile_env_vars = profile_renderer.ctx_obj.env_vars
+        return profile
+
+    @classmethod
+    def collect_parts(cls: Type["RuntimeConfig"], args: Any) -> Tuple[Project, Profile]:
+        # profile_name from the project
+        project_root = args.project_dir if args.project_dir else os.getcwd()
```

Comment:
Can we not use any `os` methods for paths?  It's the "old" filesystem interface, PathLib will save us a lot of headaches, esp when it comes to cross-platform stuff.

Reply:
I think this part of the code will go away in the new CLI? just refactored it to make the new parts easier, we are removing this line in the new code and just use [this](https://github.com/dbt-labs/dbt-core/pull/6142#discussion_r1012244317)

Reply:
And kept this function so all tests still passin